src/unit_tests/tests.rs

```diff
@@ -721,3 +721,345 @@ fn actionctx_installaddon_update_fail() {
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+        },
+        1000,
+    );
+    run(
+        runtime.dispatch(&Msg::Action(Action::Ctx(ActionCtx::UninstallAddon(
+            addon.transport_url,
+        )))),
+    );
+    assert!(
+        runtime.app.read().unwrap().ctx.profile.addons.is_empty(),
+        "addons updated successfully in memory"
+    );
+    assert!(
+        STORAGE
+            .read()
+            .unwrap()
+            .get(PROFILE_STORAGE_KEY)
+            .map_or(true, |data| {
+                serde_json::from_str::<Profile>(&data)
+                    .unwrap()
+                    .addons
+                    .is_empty()
+            }),
+        "addons updated successfully in storage"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().len(),
+        1,
+        "One request has been sent"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().get(0).unwrap().to_owned(),
+        Request {
+            url: "https://api.strem.io/api/addonCollectionSet".to_owned(),
+            method: "POST".to_owned(),
+            body: "{\"type\":\"AddonCollectionSet\",\"authKey\":\"auth_key\",\"addons\":[]}"
+                .to_owned(),
+            ..Default::default()
+        },
+        "addonCollectionSet request has been sent"
+    );
+}
+
+#[test]
+fn actionctx_uninstalladdon_uninstall_protected() {
+    #[derive(Model, Debug, Default)]
+    struct Model {
+        ctx: Ctx<Env>,
+    }
+    let addon = Descriptor {
+        manifest: Manifest {
+            id: "id".to_owned(),
+            version: Version::new(0, 0, 1),
+            name: "name".to_owned(),
+            contact_email: None,
+            description: None,
+            logo: None,
+            background: None,
+            types: vec![],
+            resources: vec![],
+            id_prefixes: None,
+            catalogs: vec![],
+            addon_catalogs: vec![],
+            behavior_hints: Default::default(),
+        },
+        transport_url: "transport_url".to_owned(),
+        flags: DescriptorFlags {
+            official: false,
+            protected: true,
+            extra: Default::default(),
+        },
+    };
+    Env::reset();
+    STORAGE.write().unwrap().insert(
+        PROFILE_STORAGE_KEY.to_owned(),
+        serde_json::to_string(&Profile {
+            addons: vec![addon.to_owned()],
+            ..Default::default()
+        })
+        .unwrap(),
+    );
+    let (runtime, _) = Runtime::<Env, Model>::new(
+        Model {
+            ctx: Ctx {
+                profile: Profile {
+                    addons: vec![addon.to_owned()],
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+        },
+        1000,
+    );
+    run(
+        runtime.dispatch(&Msg::Action(Action::Ctx(ActionCtx::UninstallAddon(
+            addon.to_owned().transport_url,
```

Comment:
that to_owned call is in a strange place. it usually put last to avoid unnecessary mem copy.
you could use hardcoded strings here